_posts/2021-01-31-choosing-randomization-unit.md

```diff
@@ -0,0 +1,353 @@
+
+* The A/B test correctly detects a positive effect 100.0% of the time (at a 95% "confidence" level)
+* The A/B test detects an effect size of 203 bps (~2%) on average
+
+When sessions are truly independent, everything works out as we'd expect.
+
+## Session level randomization with nonindependence
+
+But, as I discussed above, it is very unlikely that sessions are completely independent.
+
+<p align="center">
+    <img width="100%" src="{{ site.baseurl }}{% link images/choosing-randomization-unit/session_level_split_w_memory.png %}">
+</p>
+
+With the product recommendations experiment example I gave above in mind, let's assume that once a user experiences a session in the treatment group, their **session conversion rate is permanently increased**. Again, head to <a href="#adjusting-for-nonindependence">the appendix</a> to see how this is implemented. This time we see that:
+
+* The A/B test correctly detects a positive effect 90.0% of the time (at a 95% "confidence" level)
+* The A/B test detects an effect size of 137 bps (~1.4%) on average
+
+With this carryover effect, we're now under estimating the true effect, and detecting a positive impact less often. This may not seem too bad since we still correctly detected a positive effect. However, with smaller effect sizes, you will run into situations where this "carryover effect", the user's conversion permanently changing, causes you to miss detecting any positive effect entirely.
+
+## User level randomization
+
+Randomizing on the user instead of the session can help us deal with these non-independent sessions. In order to avoid the same pitfalls we faced with session level randomization, we must assume that the users themselves are independent, and do not influence one another's decision to convert <sup><a href="#notes">2</a></sup>. Using our faithful shoppers Terry & Jerry, user level randomization would look something like this:
+
+<p align="center">
+    <img width="100%" src="{{ site.baseurl }}{% link images/choosing-randomization-unit/user_level_split.png %}">
+</p>
+
+We can see that each user is consistently assigned to the same experience across all their sessions. After re-running our simulation with user level randomization, and the same "carryover effects" from before, we recover the results we originally expected:
+
+* The A/B test correctly detects a positive effect 100.0% of the time (at a 95% "confidence" level)
+* The A/B test detects an effect size of 199 bps (~2%) on average
+
+
```

Comment:
Could create another diagram that looks something like two bubbles - test & control - with all their individual units (sessions) and show how some sessions between the two groups are linked with lines.